### ISSUE TITLE
Clustering of unlabeled data

### ISSUE THREAD

**Issue by Gvaihir** [facebookresearch/deepcluster#27]
Hi! Great work!

I've trained AlexNet arch with deep cluster on ~1 million of unlabeled images. I want to use the trained parameters to cluster another (way bigger) set of unlabeled images. Should I use a linear classifier as in eval_linear.py or just run 1 additional epoch of training with frozen layers  and then pseudolabel extraction?
Thanks!

**Comment by mathildecaron31** [comment]
Hi,
Thanks for your interest !

I don't see how you could use a linear classifier if you don't have labels for your large set of images. What labels would you use to train the classifier ?

I would recommend that you use your deepcluster-pretrained AlexNet to extract features for the large dataset and then cluster these features.
If you can afford to run epochs of deepcluster on the large dataset, I would recommend you to do so.

Hope it helps.

**Comment by Gvaihir** [comment]
Thanks for reply! 
Are pseudolabels saved in the pickle of trained AlexNet, so that I could use model.eval() as for normal classification on those pseudo labels?
Thanks!

**Comment by mathildecaron31** [comment]
For a classification task on a supervised dataset with N classes, you can take a deepcluster-trained net, re-define the last prediction layer to a 4096*N layer and fine-tune all the network on the classification task.
In this scenario, deepcluster acts as a pre-training for your classification task.

You can also use the deepcluster-trained net to directly classify a dataset into the pseudo-classes discovered by deepcluster. I don't know if this is a good idea though.

Hope it answers your question.

**Comment by Gvaihir** [comment]
Thanks for your replies!
